WIP Launchpad: Allow user to go back to Add Links

sharpirate opened the pull request:
#### Summary
Currently, once the user lands on the Launchpad checklist screen via the LinkInBio flow, they are unable to go back to the **Add Links** task since it's disabled:

https://user-images.githubusercontent.com/20927667/189232830-28d74eee-bda0-4f50-aae1-daa6a21adac8.mov

#### Proposed changes
This PR allows users to revisit the **Add Links** task:

https://user-images.githubusercontent.com/20927667/189232848-836262a5-9dec-44c1-863f-275d2653818c.mov

#### Testing Instructions
1. Go to `http://calypso.localhost:3000/setup/launchpad?flow=link-in-bio&siteSlug=YOUR_SITE_SLUG&notice=purchase-success`
2. You should see that the **Add Links** task is completed but still active
3. Click on the **Add Links** task
4. You should see the Site Editor screen

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes #67430

edanzer committed:
Enable link to Setup step from Launchpad

edanzer committed:
Allow direct return to Launchpad from Setup step

edanzer committed:
Apply new keepActive property to Setup pages

edanzer committed:
Add return-to-launchpa logic to Link-in-bio setup

sharpirate committed:
Merge branch 'update/launchpad-return-to-setup' of github.com:Automattic/wp-calypso into add/go-back-to-add-links

sharpirate committed:
allow user to go back to add links

Sharpirate commented:
Note: I've synced this PR to [Erick's PR](https://github.com/Automattic/wp-calypso/pull/67523) due to his changes in the task-helper file so we might have to sync it again if there are any new changes on Erick's PR

jeyip commented:
Sorry for the ping folks! I tried updating the PR's base branch, but there were divergent commits.

jeyip commented:
## Testing
### Requirements
- [x] Can edit links even after link in bio task has been completed

### Browsers
- [x] Chrome

The code change and the behavior

worldomonation committed:
KPI Dashboard: add Allure reporter to Jest environment. (#67333)

* Install allure-js-commons

* Add new package @automattic/jest-circus-allure-reporter.

* Add new package `@automattic/jest-circus-allure-reporter` to dependencies throughout the tree.

* packages/calypso-e2e/src/jest-playwright-config/environment.ts
- hook the Allure reporter into our Jest environment.

* .teamcity/_self/lib/customBuildType/E2EBuildType.kt
- build package in CI

test/e2e/.gitignore
- add allure-results to git ignore

test/e2e/package.json
- define new scripts for `test/e2e` to make life easier

* .teamcity/_self/lib/customBuildType/E2EBuildType.kt
- experimental: add allure-results to the artifacts list

* packages/calypso-e2e/tsconfig.json
- add reference to `@automattic/jest-circus-allure-reporter`.

packages/tsconfig.json
- fix alphabetical ordering

* .teamcity/_self/lib/customBuildType/E2EBuildType.kt
- in the collect logs step add referemces tp allure-results

packages/calypso-e2e/src/env-variables.ts
- add new environment variable ALLURE_RESULTS_PATH

packages/calypso-e2e/src/types/env-variables.types.ts
- add ALLURE_RESULTS_PATH to typng

packages/calypso-e2e/src/jest-playwright-config/environment.ts
- use ALLURE_RESULTS_PATH when configuring allure

* .teamcity/_self/lib/customBuildType/E2EBuildType.kt
- fix folder name

* .teamcity/_self/lib/customBuildType/E2EBuildType.kt
- remove experimental section on adding support for allure-results output

* packages/calypso-e2e/src/env-variables.ts
- change default value for ALLURE_RESULTS_PATH to an empty string.

packages/calypso-e2e/src/jest-playwright-config/environment.ts
- short circuit the allure reporter setup step in `initializeAllureReporter` if `ALLURE_RESULTS_PATH` is not set.
- since `this.allure` attribute can be undefined, replace calls to `this.allure` with `this.allure?`.

arcangelini committed:
HelpCenter: improve window state management with HappyChat (#67353)

* Working model

* Cleanup HappyChat parent connection

* Added control for close/end chat

* Re-add comments

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

benzhu56 committed:
update wording (#67626)

Co-authored-by: Ben Zhu <ben.zhu@a8c.com>

fushar committed:
Style selection: Implement button to submit selected design and variation (#67504)

* Wrap optional params into object

miksansegundo committed:
Pattern Assembler: Pattern selector features (#67478)

* Basic layout and simple pattern selector

* Basic layout and simple pattern selector

* Add section patterns list for testing

* Preloading list of patterns

* Pattern add, replace and remove from list

* fix back to remove

* fix back to remove

* fix remove

* Remove choose button

* Dynamic pattern preview height

* fix patterns list

* Handle unmount

* Handle locale

* Use calypso_token in preview auto height

* Fix pattern ID

* Code improvements

* Code improvements

yashwin committed:
Remove chevron icon from agency dashboard actions (#67587)

taipeicoder committed:
Style selection: Show old preview for themes that don't have style variations (#67629)

sejas committed:
Site Switcher: update the new site button to show it at the bottom of the sidebar (#67322)

* Add new `transparent` style to `@automattic/components` to use it on the SiteSwitcher and the Sidebar.
* Add the new button styles to storybook.
* Add new eslint rule to package/components to fix an error on stories.
* Refactor Site Switcher `manage sites` button to use the new button style.
* Keep the current behavior and show the button for users with only one site.
* Show this button at the bottom of the calypso sidebar, admin menu.
* Add plus icon for collapse mode, check the gif below.

bogiii committed:
Turn on the subscriber role when the subscriber/importer feature flag is off (#67639)

Closes #67531

flootr committed:
`HappychatSupport`: refactor tests to `@testing-library/react` (#67540)

* `HappychatSupport`: refactor tests to `@testing-library/react`

* remove redundant `true` prop assignments

* use `get*` instead of `query*` where applicable

manzoorwanijk committed:
Pricing page rework: Prevent interaction with disabled buttons (#67588)

* Prevent interaction with disabled buttons

* Fix disabled button background inside lightbox on WPCOM

olaseni committed:
Google Workspace Regional Discount: Add new pricing information for products with an introductory offer (#67408)

* Add new pricing information for GW products with an introductory offer

* Review suggestions

* Reformat messaging

* Improve messaging

escapemanuele committed:
Tailored Flows: Multiline description in setup step (#67608)

epeicher committed:
Marketplace search: Use plugin.author.raw and author_name to filter by author (#67327)

* Marketplace search: Use author.raw and author_name to filter by author

* Add plugin.author.raw as well as plugin.author to search filter

This will be used when searching by author

* Use only plugin.author.raw for filtering

The plugin.author criteria is not required

j6ll committed:
Promote posts design updates (#67443)

* Colour of the anchor does not match the design
- fix linting warning as the border radius makes the item circlular (rather than rounded corner)

* Adds subtitle and support link article

Co-authored-by: James Gill <james.gill@automattic.com>

jjolmo committed:
Integrate first pass with BlazePress translations (#67594)

Co-authored-by: Javier Olmo <javier.olmo@automattic.com>

andres-blanco committed:
remove Advertising option is not enabled (#67056)

leonardost committed:
Update dot color of expiring domain connections to green (#67621)

alshakero committed:
Tailored Flows: empty the store after submitting info (#67567)

alshakero committed:
Tailored flows: Link in Bio - pattern Picker: fix dragging spill in pattern picker (#67681)

arcangelini committed:
Make CTA contact form submit fixed (#67625)

arcangelini committed:
Add logic for helpcenter in HappyChat connection (#67560)

john-legg committed:
Updating Playwright version to 1.23 (#67696)

danielbachhuber committed:
Sites Management Page: Use consistent casing for launch status dropdown (#67697)

zaguiini committed:
Sites Management Dashboard: Add dropdown to choose between alphabetical and last publish sorting (#67689)

Co-authored-by: Daniel Bachhuber <daniel.bachhuber@automattic.com>

fushar committed:
Style selection: Move the submit button to bottom-left on desktop resolution (#67642)

benzhu56 committed:
Change position of user name and post type (#67474)

Co-authored-by: Ben Zhu <ben.zhu@a8c.com>

benzhu56 committed:
update wording for cancel/stop campaign (#67627)

Co-authored-by: Ben Zhu <ben.zhu@a8c.com>

jkguidaven committed:
Pricing page rework: Enable compact mode for Item price component when viewed in tight space. (#67634)

* Break time-frame component text with span element.

* Create useItemPriceCompact hook to handle item price compact state.

* Remove listener when containerRef changes in useItemPriceCompact

* Display normal and compact translation into separate blocks for better translation.

* Add comment for translator in the new compact string.

yashwin committed:
Plugin Management: UI enhancements based on CFT feedback (#67637)

* Update color of "updated x days ago" text

* Disabled individual plugin update when bulk action is active

* Update styling of disabled bulk action buttons

* Remove "Actions" dropdown option in the mobile view as it is redundant

* Fit last updated column text into single line

* Add max-width only on wide screens since that was causing issues

jeyip committed:
Launchpad: Reintroduce loading screen styles (#67700)

taipeicoder committed:
Style selection: Style preview UI/UX polish (#67436)

candy02058912 committed:
Marketplace: Add empty page for plans (#67710)

* Marketplace: Add empty page for plans

* Select site if missing site

* Couldn't get it to work, lets just force a site for now

* Remove nav header

* Page tracker path / title

* Revert "Remove nav header"

This reverts commit 719e1b1fc7d00744c1a4e81d49d837bac02c3967.

* Use fixed nav instead

* Remove route localization

* Change to tsx

Co-authored-by: Louis Laugesen <louis.laugesen@gmail.com>

flootr committed:
`ProductPurchaseFeaturesList`: refactor tests to `@testing-library/react` (#66966)

* `ProductPurchaseFeaturesList`: refactor tests to `@testing-library/react`

* use `get*` instead of `query*` where applicable

* Remove redundant `={ true }`

* Fix indentation

* remove `screen.debug()` call

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

vikneshwar committed:
Add additional product data(lightbox) to calypso package (#67687)

* Jetpack New Pricing Page - Add additional product data to calypso package

* Adds lightbox additional data - Includes, Recommended for, Benefits to calypso-products package
* Use the data from calypso package to generate item slug

* Jetpack New Pricing Page - Remove constants for translation
* Remove constant that stored the translated text, instead call translate inline
* Remove redundant type cast and return types

* Jetpack New Pricing Page - Removed extra benefits

* Jetpack New Pricing Page - refactor translations
* Remove extra whitespace in translation texts
* Rename selector function

bogiii committed:
Subscribers form: integrate record tracks events (#67555)

* Introduce record add form events hook

* Integrate recordAddFormEvents hook and fix bug with clearing import error state

* Integrate recordTrackEvents with AddSubscriberForm usage

* Remove obsolete progress bar simulation

arthur791004 committed:
Pattern Assembler: Go back to the main screen when clicking on the “Back” button (#67638)

* Pattern Assembler: Go back to the main screen when clicking on the Back button

* Fix position of Continue button

escapemanuele committed:
Tailored Flows: Leaving checkout brings you to launchpad (#67672)

escapemanuele committed:
Tailored flow: Fix line height problem in setup textarea (#67676)

escapemanuele committed:
Tailored Flow (1 min review): Link in bio intro text to set up (#67718)

claudiucelfilip committed:
Ensure all breadcrumb items get reconstructed after the translations get fetched (#67431)

alshakero committed:
WP data stores: fix store persistence misconfiguration and add selectors for siteTitle, logo and description (#67719)

yashwin committed:
Plugin Management: UI enhancements based on CFT feedback - part 2 (#67724)

* Show plugin count after it is fetched instead of showing "0"

* Reduce vertical margin on "Installed on X sites" on card(small screen) view

* Update empty state for plugin avatar / image

renancarvalho committed:
Add/high quality background image (#67728)

* Add LiB high quality images

* Add Newsletter high quality images

* Change breakpoint for larger images

heavyweight committed:
NewsletterSetup: remove outline on focus of Accent Color Popover (#67675)

Nic-Sevic committed:
Update plugin-compatibility.js (#66050)

Co-authored-by: Daniel Bachhuber <daniel.bachhuber@automattic.com>

danielbachhuber committed:
Sites Management Page: Improve render performance by paginating when greater than 96 sites (#67690)

gmovr committed:
Google Analytics 4 For WPcom  (#65735)

* Added measurement ID for WordPress.com, and initializing this via window.gtag. Some tests to make sure we're intializing WPcom and Jetpack gtags correctly. Started to refactor some Jetpack related code to work on WPcom as well.

* Removed unneeded extra mock/spy in tests

* Always init wpcom gtag

* Started on WPcom purchase tracking, made functions for cart more generic ... Still need to refactor out duplicate code between Jetpack and WPcom

* Refactor functions to be used on WPcom and Jetpack

* Adding enum type for tracking environment, and testing implementation of this on GA4 page views and purchases

* Removing extra property for trackingEnvironment/sendToJetpack. This will be returned form cartToGaPurchase instead, making the implementation cleaner

* Parsing the brand from the cart, instead of relying on the tracking environment to set the product brand. This is because the current implementation of Enhanced Ecommerce (Universal Analytics) on WPcom also fires on Jetpack purchases on WPcom.

* Changing the name of fireJetpackEcommerceAddToCart, to a more generic name, as it will also be implemented on WPcom

* Fix page view bug for looking up Gtag ID with enum

* Fixing wrong gtag on recordOrderInJetpackGA

* If the cart contains a WordPress.com product, also add any (potential) Jetpack product in the cart to the cart items. This is an edge case, but there may be cases where someone has _both_ a plan and Jetpack search in their cart. If the cart only contains a Jetpack product (whether it's on WPcom or Jetpack Live, the behavior is the same as previously

* Fixing wrong lookup for tracking ID

* Adding additional assertion for GA4 tests

leonardost committed:
Domains: Remove unused properties from the `resolveDomainStatus` function (#67671)

* Remove unused `listStatusClass` property from the `resolveDomainStatus` function`

* Remove unused `listStatusText` property

leonardost committed:
Domains: remove unused domain components (#67682)

* Remove unused `TransferIn` and `SiteRedirect` components

* Remove unused import of `DnsRecordsList`

* Remove unused EditContactInfo component

leonardost committed:
Redirect the `/domain/manage/all/edit-contact-info` route (#67701)

That route was added in #53481 but led to a broken page because
`DomainManagement.BulkEditContactInfo` wasn't defined (see
https://github.com/Automattic/wp-calypso/blob/1ff4a6d7eb8a9f1065b6d9649674bf7e5ae2a318/client/my-sites/domains/domain-management/index.jsx).

This commit redirects that route to `/domains/manage?site=all&action=edit-contact-email`
until we implement the original functionality of bulk editing contact
info (currently we only update emails in bulk).

gcsecsey committed:
update `pluginRequiresCustomPrimaryDomain` to use `seo` tag instead of the `required_primary_domain` field (#67734)

renancarvalho committed:
Fix header text on domain/plans step (#67600)

* Fix header text on domain/plans step

* Fix header title  and remove wrong style

alshakero committed:
Tailored flows: Fix pattern picker in RTL (#67609)

sharpirate committed:
recreate changes off of trunk

sharpirate committed:
remove unused imports
